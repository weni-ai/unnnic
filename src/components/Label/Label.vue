<template>
  <p class="unnnic-label__label">{{ label }}</p>
</template>

<script>
export default {
  name: 'unnnic-label',
  props: {
    label: {
      type: String,
      default: null,
    },
  },
};
</script>

<style lang="scss">
@import "../../assets/scss/unnnic.scss";

.unnnic-label {
  &__label {
    font-family: $unnnic-font-family-secondary;
    font-weight: $unnnic-font-weight-regular;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-medium;
    font-size: $unnnic-font-size-body-gt;
    color: $unnnic-color-neutral-cloudy;
    margin: $unnnic-spacing-stack-xs 0;
  }
}
</style>
