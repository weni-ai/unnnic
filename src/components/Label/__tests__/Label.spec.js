import { mount } from '@vue/test-utils';

import Label from '../Label.vue';
import { beforeEach, describe, expect, it } from 'vitest';

const createWrapper = () => {
  return mount(Label, {
    props: {
      label: 'Label Test',
    },
  });
};

describe('Label', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = createWrapper();
  });

  it('should render label', () => {
    const label = wrapper.findComponent({ name: 'UnnnicLabel' });
    expect(label.text()).eq('Label Test');
  });

  it('applies the correct classes and styles', () => {
    const label = wrapper.find('.unnnic-label__label');
    expect(label.html()).toMatchSnapshot();
  });
});
