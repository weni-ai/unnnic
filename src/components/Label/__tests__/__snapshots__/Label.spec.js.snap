// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label > applies the correct classes and styles 1`] = `"<p data-v-7f222291="" class="unnnic-label__label">Label Test</p>"`;
